src/components/WalletModal/index.tsx

```diff
@@ -126,101 +119,109 @@ export default function WalletModal({
+  const [pendingConnector, setPendingConnector] = useState<Connector | undefined>()
+  const pendingError = useAppSelector((state) =>
+    pendingConnector ? state.wallet.errorByWallet[getWalletForConnector(pendingConnector)] : undefined
+  )
 
   const walletModalOpen = useModalOpen(ApplicationModal.WALLET)
   const toggleWalletModal = useWalletModalToggle()
 
-  const previousAccount = usePrevious(account)
+  const openOptions = useCallback(() => {
+    setWalletView(WALLET_VIEWS.OPTIONS)
+  }, [setWalletView])
 
-  const resetAccountView = useCallback(() => {
-    setPendingError(false)
-    setWalletView(WALLET_VIEWS.ACCOUNT)
-  }, [setPendingError, setWalletView])
-
-  // close on connection, when logged out before
   useEffect(() => {
-    if (account && !previousAccount && walletModalOpen) {
-      toggleWalletModal()
+    if (walletModalOpen) {
+      setWalletView(connector === network ? WALLET_VIEWS.OPTIONS : WALLET_VIEWS.ACCOUNT)
     }
-  }, [account, previousAccount, toggleWalletModal, walletModalOpen])
+  }, [walletModalOpen, setWalletView, connector])
 
-  // always reset to account view
   useEffect(() => {
-    if (walletModalOpen) {
-      resetAccountView()
+    if (pendingConnector && walletView !== WALLET_VIEWS.PENDING) {
+      updateWalletError({ wallet: getWalletForConnector(pendingConnector), error: undefined })
+      setPendingConnector(undefined)
     }
-  }, [walletModalOpen, resetAccountView])
+  }, [pendingConnector, walletView])
+
+  const tryActivation = useCallback(
+    async (connector: Connector) => {
+      const name = Object.values(SUPPORTED_WALLETS).find(
+        (supportedWallet) => connector === supportedWallet.connector
+      )?.name
+
+      // log selected wallet
+      sendEvent({
+        category: 'Wallet',
+        action: 'Change Wallet',
+        label: name,
+      })
 
-  const tryActivation = async (connector: AbstractConnector | undefined) => {
-    let name = ''
-    Object.keys(SUPPORTED_WALLETS).map((key) => {
-      if (connector === SUPPORTED_WALLETS[key].connector) {
-        return (name = SUPPORTED_WALLETS[key].name)
-      }
-      return true
-    })
-    // log selected wallet
-    sendEvent({
-      category: 'Wallet',
-      action: 'Change Wallet',
-      label: name,
-    })
-    setPendingWallet(connector) // set wallet for pending view
-    setWalletView(WALLET_VIEWS.PENDING)
+      const wallet = getWalletForConnector(connector)
 
-    // if the connector is walletconnect and the user has already tried to connect, manually reset the connector
-    if (connector instanceof WalletConnectConnector) {
-      connector.walletConnectProvider = undefined
-    }
+      try {
+        // Fortmatic opens it's own modal on activation to log in. This modal has a tabIndex
+        // collision into the WalletModal, so we special case by closing the modal.
+        if (connector === fortmatic) {
+          toggleWalletModal()
```

Comment:
Replace this with the old logic so there is not a delay between closing our modal and opening theirs.

```
fortmatic.on(OVERLAY_READY, toggleWalletModal)
```

Reply:
this adds so much logic that i found confusing when i was learning the codebase that i think the delay is okay... especially with how few people even use fortmatic. longer term i think we should deprecate fortmatic anyways